---
title: "Academic CV"
date: 2023-10-17T13:27:01Z
draft: false
---
> <button type="button" name="resume-switcher" onclick="location.href='/resume'">Switch to Resume</button>
> &ZeroWidthSpace;  
> &ZeroWidthSpace;[Download PDF](/documents/academic_cv_tommi_maklin.pdf)  
> &ZeroWidthSpace;<script src="/js/contact_me.js"></script><noscript>[Turn on JavaScript to see my email address.]</noscript>  
> &ZeroWidthSpace;<script src="/js/call_me.js"></script><noscript>[Turn on JavaScript to see my phone number.]</noscript>  

------

### Education {#education}
&nbsp;
__University of Helsinki, Helsinki, Finland__

: PhD, Computer Science, 2022   
Probabilistic methods for high-resolution metagenomics
<br><br>

__University of Helsinki, Helsinki, Finland__

: MSc, Statistical machine learning, 2017  
Major studies in Statistics
<br><br>

__University of Helsinki, Helsinki, Finland__

: BSc, Statistics, 2016  
Minor studies in Mathematics and Computer Science

------

### Research experience {#employment}
&nbsp;
__University of Helsinki, Helsinki, Finland__
: Postdoctoral researcher, 2023--ongoing  

: I work on developing and applying statistical methods for studying the
link between ecology and epidemiology of opportunistic pathogens that
are also a part of the healthy human gut microbiome.
<br><br>

__European Bioinformatics Institute, Hinxton, Cambridgeshire, UK__
: Visiting postdoctoral fellow, Jan 2023--July 2023  

: I visited the Pathogen Informatics and Modelling group led by
Dr John Lees through the Theory@EMBL program.
<br><br>

__University of Helsinki, Helsinki, Finland__
: Doctoral researcher, 2018--2022  

: Developed probabilistic models for determining the composition of a
DNA sequencing data sample. Produced efficient open source
implementations of the said models and distributed them. Published
results in scientific journals and presented at international
conferences.
<br><br>

__Helsinki Institute for Information Technology, Helsinki, Finland__
: Research assistant, 2016--2017  

: Designed and implemented models for identifying bacterial strains from DNA sequencing data.

-------

### Other employment {#experience}
&nbsp;
__Heldata oy, Helsinki, Finland__
: Part-time entrepreneur, 2018--2022   

: Planned, organised, and hosted a 250-person data science networking
and recruiting event in 2019 and 2021. Ran the day-to-day affairs of
the business. Worked with customers to find bespoke solutions.
<br><br>

__Stokastiko oy, Helsinki, Finland__
: Independent consultant, 2016--2018  

: Designed and taught a 3-day statistics introductory course for the tax
avoidance unit of the Finnish Tax Administration. Consulted small
businesses and organizations in automating data analysis
pipelines. Founded and operated a one-man company.
<br><br>

-------

### Publications {#publications}
&nbsp;
Preprints

: Group 2 and 3 ABC-transporter dependant capsular K-loci contribute significantly to variation in the invasive potential of _Escherichia coli_  
RA&nbsp;Gladstone, M&nbsp;Pesonen, AK&nbsp;P&ouml;ntinen, __T&nbsp;M&auml;klin__, N&nbsp;MacAlasdair, H&nbsp;Thorpe, Y&nbsp;Shao, S&nbsp;Mallawaarachchi, S&nbsp;Arredondo-Alonso, BJ&nbsp;Parcell, JD&nbsp;Turnbull, G&nbsp;Tonkin-Hill, PJ&nbsp;Johnsen, &Oslash;&nbsp;Samuelsen, NR&nbsp;Thomson, TD&nbsp;Lawley, and&nbsp;J&nbsp;Corander  
*preprint* (2024), available at *medRxiv*: [10.1101/2024.11.22.24317484](https://doi.org/10.1101/2024.11.22.24317484)

: Enhanced metagenomics-enabled transmission inference with TRACS  
G&nbsp;Tonkin-Hill, Y&nbsp;Shao, AE&nbsp;Zarebski, S&nbsp;Mallawaarachchi, Ouli&nbsp;Xie, __T&nbsp;M&auml;klin__, HA&nbsp;Thorpe, MR&nbsp;Davies, SD&nbsp;Bentley, TD&nbsp;Lawley, and&nbsp;J&nbsp;Corander  
*preprint* (2024), available at *bioRxiv*: [10.1101/2024.08.19.608527](https://doi.org/10.1101/2024.08.19.608527)  
software in GitHub: [TRACS](https://github.com/gtonkinhill/tracs)

: Basic reproduction number for pandemic _Escherichia coli_ clones is comparable to typical pandemic viruses  
FA&nbsp;Ojala, H&nbsp;Pesonen, RA&nbsp;Gladstone, __T&nbsp;M&auml;klin__, G&nbsp;Tonkin-Hill, P&nbsp;Marttinen, and&nbsp;J&nbsp;Corander  
*preprint* (2024), available at *bioRxiv*: [10.1101/2024.05.08.593267](https://doi.org/10.1101/2024.05.08.593267)

<br>

Peer-reviewed journal articles

: Geographical variation in the incidence of colorectal cancer and urinary tract cancer is associated with population exposure to colibactin-producing _Escherichia coli_  
__T&nbsp;M&auml;klin__, A&nbsp;Taira, S&nbsp;Arredondo-Alonso, Y&nbsp;Shao, MR&nbsp;Stratton, TD&nbsp;Lawley, LA&nbsp;Aaltonen, and&nbsp;J&nbsp;Corander  
*The Lancet Microbe* (2024), doi: [10.1016/j.lanmic.2024.101015](https://doi.org/10.1016/j.lanmic.2024.101015)

: Seamless, rapid, and accurate analyses of outbreak genomic data using split _k_-mer analysis  
R&nbsp;Derelle, J&nbsp;von&nbsp;Wachsmann, __T&nbsp;M&auml;klin__, J&nbsp;Hellewell, T&nbsp;Russell, A&nbsp;Lalvani, L&nbsp;Chindelevitch, NJ&nbsp;Croucher, SR&nbsp;Harris, and&nbsp;JA&nbsp;Lees  
*Genome Research* (2024), doi: [10.1101/gr.279449.124](https://doi.org/10.1101/gr.279449.124)  
software in GitHub: [SKA2](https://github.com/bacpop/ska.rust)

: Pan-pathogen deep sequencing of nosocomial bacterial pathogens in Italy in Spring 2020: a prospective cohort study  
HA&nbsp;Thorpe, M&nbsp;Pesonen, M&nbsp;Corbella, H&nbsp;Pesonen, S&nbsp;Gaiarsa, CJ&nbsp;Boinett, G&nbsp;Tonkin-Hill, __T&nbsp;M&auml;klin__, AK&nbsp;P&ouml;ntinen, N&nbsp;MacAlasdair, RA&nbsp;Gladstone, S&nbsp;Arredondo-Alonso, T&nbsp;Kallonen, D&nbsp;Jamrozy, SW&nbsp;Lo, C&nbsp;Chaguza, GA&nbsp;Blackwell, A&nbsp;Honkela, AC&nbsp;Sch&uuml;rch, RJL&nbsp;Willems, C&nbsp;Merla, G&nbsp;Petazzoni, EJ&nbsp;Feil, P&nbsp;Cambieri, NR&nbsp;Thomson, SD&nbsp;Bentley, D&nbsp;Sassera, and&nbsp;J&nbsp;Corander  
*The Lancet Microbe* (2024), doi: [10.1016/S2666-5247(24)00113-7](https://doi.org/10.1016/S2666-5247(24)00113-7)

: Deep sequencing of _Escherichia coli_ exposes colonisation diversity and impact of antibiotics in Punjab, Pakistan  
T&nbsp;Khawaja&#176;, __T&nbsp;M&auml;klin&#176;__, T&nbsp;Kallonen&#176;, RA&nbsp;Gladstone, AK&nbsp;P&ouml;ntinen, S&nbsp;Mero, HA&nbsp;Thorpe, &Oslash;&nbsp;Samuelsen, J&nbsp;Parkhill, M&nbsp;Izhar, WM&nbsp;Akhtar, J&nbsp;Corander, and&nbsp;A&nbsp;Kantele  
&#176; joint first authors  
*Nature Communications* (2024), doi: [10.1038/s41467-024-49591-5](https://dx.doi.org/10.1038/s41467-024-49591-5)

: Exploring the paediatric nasopharyngeal bacterial microbiota with culture-based MALDI-TOF mass spectrometry and targeted metagenomic sequencing  
P&nbsp;Sreymom, T&nbsp;Kallonen, __T&nbsp;M&auml;klin__, P&nbsp;Sar, J&nbsp;Hopkins, S&nbsp;Soeng, T&nbsp;Miliya, CL&nbsp;Ling, SD&nbsp;Bentley, J&nbsp;Corander, and&nbsp;P&nbsp;Turner  
*mBio* (2024), doi: [10.1128/mbio.00784-24](https://doi.org/10.1128/mbio.00784-24)

: A novel prophage-like insertion element within yabG triggers early entry into sporulation in _Clostridium botulinum_  
FP&nbsp;Douillard, I&nbsp;Martins&nbsp;Portinha, Y&nbsp;Derman, C&nbsp;Woudstra, __T&nbsp;M&auml;klin__, MB&nbsp;Dorner, H&nbsp;Korkeala, AO&nbsp;Henriques, and&nbsp;M&nbsp;Lindstr&ouml;m  
*Viruses* (2023), doi: [10.3390/v15122431](https://doi.org/10.3390/v15122431)

: Themisto: a scalable colored _k_-mer index for sensitive pseudoalignment against
hundreds of thousands of bacterial genomes  
JN&nbsp;Alanko, J&nbsp;Vuohtoniemi, __T&nbsp;M&auml;klin__, and&nbsp;SJ&nbsp;Puglisi  
*Bioinformatics* (2023), doi: [10.1093/bioinformatics/btad233](https://doi.org/10.1093/bioinformatics/btad233)  
software in GitHub: [Themisto](https://github.com/algbio/themisto)

: Strong pathogen competition in neonatal gut colonisation  
__T&nbsp;M&auml;klin__, HA&nbsp;Thorpe, AK&nbsp;P&ouml;ntinen, RA&nbsp;Gladstone, Y&nbsp;Shao, M&nbsp;Pesonen, A&nbsp;McNally, PJ&nbsp;Johnsen, &Oslash;&nbsp;Samuelsen, TD&nbsp;Lawley, A&nbsp;Honkela, and&nbsp;J&nbsp;Corander  
*Nature communications* (2022), doi: [10.1038/s41467-022-35178-5](https://doi.org/10.1038/s41467-022-35178-5)

: Genomic and phenotypic characterization of Clostridium botulinum isolates from an infant botulism case suggests adaptation signatures to the gut  
FP&nbsp;Douillard, Y&nbsp;Derman, C&nbsp;Woudstra, K&nbsp;Selby, __T&nbsp;M&auml;klin__, MB&nbsp;Dorner, H&nbsp;Sax&eacute;n, BG&nbsp;Dorner, H&nbsp;Korkeala, and&nbsp;M&nbsp;Lindstr&ouml;m  
*mBio* (2022), doi: [10.1128/mbio.02384-21](https://doi.org/10.1128/mbio.02384-21)

: Bacterial genomic epidemiology with mixed samples  
__T&nbsp;M&auml;klin__, T&nbsp;Kallonen, JN&nbsp;Alanko, &Oslash;&nbsp;Samuelsen, K&nbsp;Hegstad, V&nbsp;M&auml;kinen, J&nbsp;Corander, E&nbsp;Heinz, and&nbsp;A&nbsp;Honkela  
*Microbial genomics* (2021), doi: [10.1099/mgen.0.000691](https://doi.org/10.1099/mgen.0.000691.)  
software in GitHub: [mGEMS](https://github.com/probic/mgems)

: High-resolution sweep metagenomics using ultrafast read mapping and inference  
__T&nbsp;M&auml;klin__, T&nbsp;Kallonen, S&nbsp;David, CJ&nbsp;Boinett, B&nbsp;Pascoe, G&nbsp;M&eacute;ric, DM&nbsp;Aanensen, EJ&nbsp;Feil, S&nbsp;Baker, J&nbsp;Parkhill, SK&nbsp;Sheppard, J&nbsp;Corander, and&nbsp;A&nbsp;Honkela  
*Wellcome open research* (2021), doi: [10.12688/wellcomeopenres.15639.2](https://doi.org/10.12688/wellcomeopenres.15639.2)  
software in GitHub: [mSWEEP](https://github.com/probic/msweep)

: Closing _Clostridium botulinum_ Group III genomes using long-read sequencing  
C&nbsp;Woudstra, __T&nbsp;M&auml;klin__, Y&nbsp;Derman, L&nbsp;Bano, H&nbsp;Skarin, C&nbsp;Mazuet, A&nbsp;Honkela, and&nbsp;M&nbsp;Lindstr&ouml;m  
*Microbiology resource announcements* (2021), doi: [10.1128/MRA.01364-20](https://doi.org/10.1128/MRA.01364-20)
<br><br>

Book chapters

: Identifying bacterial strains from sequencing data  
__T&nbsp;M&auml;klin__, J&nbsp;Corander, and&nbsp;A&nbsp;Honkela  
*Data mining for systems biology* (2018), doi: [10.1007/978-1-4939-8561-6_1](https://doi.org/10.1007/978-1-4939-8561-6_1)

-----

### Grants {#grants}
&nbsp;
: __Theory@EMBL visitor fellowship__  
For visit costs, approx. &pound;5&nbsp;000, 2023

: __DONASCI thesis finalization grant__  
Covered six months salary, approx. &euro;20&nbsp;000, 2021

-----

### Research visits {#researchvisits}
&nbsp;
: __Institute of Basic Medical Sciences, University of Oslo, Oslo, Norway__  
Visitor, May 2024

: __European Bioinformatics Institute, Hinxton, Cambridge, UK__  
Visiting postdoctoral fellow, Jan 2023--July 2023

-----

### External affiliations {#affiliations}
&nbsp;
: __Wellcome Sanger Institute, Hinxton, Cambridge, UK__  
Visiting worker, Pathogens and microbes programme, 2022-2024

-----

<!-- ### Teaching experience {#teaching} -->


### Positions of trust {#positionsoftrust}
&nbsp;
: __University of Helsinki, Helsinki, Finland__  
Union representative (shop steward), 2023--2024

-----

### Volunteer experience {#volunteering}
&nbsp;
: __Finnish union of university researchers and teachers__  
Board member, 2024 &SmallCircle; Deputy board member, 2023

: __Helsinki university researchers and teachers association__  
Vice chair, 2023--2024 &SmallCircle; Board member, 2022--2025 &SmallCircle; Deputy board member 2020--2021

: __University of Helsinki PhD students association__  
Chair, 2020--2021 &SmallCircle; Secretary 2022

: __Finnish statistical society__  
Secretary, 2017--2022 &SmallCircle; Deputy board member 2022--2023

-----

### Service {#service}
Boards and committees

: Doctoral school in natural sciences, University of Helsinki  
Board member, 2020--2021

: Doctoral programme in Computer Science, University of Helsinki  
Board member, 2020--2021
<br><br>

Peer review

: 2024 ECML PKDD, *Journal of Open Source Software, Microbiome*  
2023 *Genome biology, Nature communications, Microbiome, Scientific reports*  
2022 *BMC microbiology*  
2021 International conference on machine learning (ICML)  
2019 International conference on machine learning (ICML)

-----

### Research software
&nbsp;
: __mGEMS__  
*Main contributor*  
Taxonomic binner with lineage-level resolution, [PROBIC/mGEMS](https://github.com/PROBIC/mGEMS)

: __mSWEEP__  
*Main contributor*  
Taxonomic profiler with lineage-level resolution, [PROBIC/mSWEEP](https://github.com/PROBIC/mSWEEP)

: __Themisto__  
*With Jarno N Alanko & others*  
Pseudoaligner and _k_-mer index for short-read sequencing data, [algbio/Themisto](https://github.com/algbio/Themisto)

: __SKA2__  
*With John A Lees & others*  
Split _k_-mer analysis of closely related genomic sequences, [bacpop/ska.rust](https://github.com/bacpop/ska.rust)

: __cpprate__  
*Main contributor*  
Variable selection in Bayesian regression models with relative centrality, [tmaklin/cpprate](https://github.com/tmaklin/cpprate)

-----

### Media coverage {#mediacoverage}
Outlets with national reach

: Jopa puolet suolistosy&ouml;vist&auml; johtuu bakteeritartunnasta – tuore tutkimus antaa toivoa my&ouml;s rokotteesta sy&ouml;p&auml;&auml;n  
*Yle*, Helsinki, Finland, 5 December 2024  
[https://yle.fi/a/74-20129530](https://yle.fi/a/74-20129530)

: Sairaalabakteeri ei tartu terveisiin vastasyntyneisiin, kertoo tutkimus  
*Helsingin Sanomat*, Helsinki, Finland, 8 February 2023  
[https://www.hs.fi/helsinki/art-2000009380665.html](https://www.hs.fi/helsinki/art-2000009380665.html)

-----

### Presentations and conference activity {#presentations}
&nbsp;
Session chair

: Probabilistic modelling in microbial ecology workshop  
Kellogg College, University of Oxford, Oxford, UK, March 2023  
<br><br>

Invited talks

: Computational methods in AMR research  
University of Turku, Turku, Finland, October 2023

: Lawley group meeting  
Wellcome Sanger Institute, Hinxton, Cambridge, UK, June 2023

: Theory@EMBL seminar  
European Molecular Biology Laboratory EMBL, Hinxton, Cambridge, UK, June 2023

: Heinz group meeting  
Liverpool school of tropical medicine, Liverpool, UK, February 2023

: Lahti group meeting  
University of Turku, Turku, Finland, January 2023

: AMR-bridge annual meeting  
AMR-bridge consortium, Troms&oslash;, Norway, June 2022

: Thomson group meeting  
Wellcome Sanger Institute, Hinxton, Cambridge, UK, June 2020

: CompLifeSci 4th annual meeting  
&Aring;bo Akademi, Turku, Finland, November 2019
<br><br>

Contributed talks

: SMBE Satellite Workshop on Pathogen Pangenome Evolution Between and Within their Hosts
Tokyo, Japan, November 2024

: Microbial (gen)omics meeting  
Wellcome Sanger Institute, Hinxton, Cambridge, UK, July 2023

: EBI research retreat  
European Bioinformatics Institute, Hinxton, Cambridge, UK, July 2023

: ISMB/ECCB 2019  
Basel, Switzerland, July 2019

: Bioinformatics research and education workshop  
Max Planck Institute for Molecular Genetics, Berlin, Germany, May 2019

: Mathematical and statistical aspects of molecular biology workshop  
European Bioinformatics Institute, Hinxton, Cambridge, UK, April 2019

: Probabilistic modelling in genomics  
Cold Spring Harbor Laboratory, Cold Spring Harbor, NY, US, November 2018

: Regional workshop of European young researchers in Statistics  
Henri Poincar&eacute; Institute, Paris, France, November 2018

: Mathematical and statistical aspects of molecular biology workshop  
University of St Andrews, St Andrews, Scotland, UK, March 2018

-----

### Contact {#footer}
Tommi M&auml;klin -- <script src="/js/contact_me.js"></script><noscript>[Turn on JavaScript to see my email address.]</noscript> -- <script src="/js/call_me.js"></script><noscript>[Turn on JavaScript to see my phone number.]</noscript> -- [<i class="fab fa-linkedin-in fa-1x text-muted"></i>](https://linkedin.com/in/tmaklin)

------
