---
title: "Academic CV"
date: 2023-10-17T13:27:01Z
draft: false
---
> <button type="button" name="resume-switcher" onclick="location.href='/resume'">Switch to Resume</button>
> &ZeroWidthSpace;  
> &ZeroWidthSpace;[Download PDF](/documents/academic_cv_tommi_maklin_2023_10_18.pdf)  
> &ZeroWidthSpace;<script src="/js/contact_me.js"></script><noscript>[Turn on JavaScript to see my email address.]</noscript>  
> &ZeroWidthSpace;<script src="/js/call_me.js"></script><noscript>[Turn on JavaScript to see my phone number.]</noscript>  

------

### Education {#education}
&nbsp;
__University of Helsinki, Helsinki, Finland__

: PhD, Computer Science, 2022   
Probabilistic methods for high-resolution metagenomics
<br><br>

__University of Helsinki, Helsinki, Finland__

: MSc, Statistical machine learning, 2017  
Major studies in Statistics
<br><br>

__University of Helsinki, Helsinki, Finland__

: BSc, Statistics, 2016  
Minor studies in Mathematics and Computer Science

------

### Research experience {#employment}
&nbsp;
__University of Helsinki, Helsinki, Finland__
: Postdoctoral researcher, 2023--ongoing  

: I work on developing and applying statistical methods for studying the
link between ecology and epidemiology of opportunistic pathogens that
are also a part of the healthy human gut microbiome.
<br><br>

__European Bioinformatics Institute, Hinxton, Cambridgeshire, UK__
: Visiting postdoctoral fellow, Jan 2023--July 2023  

: I visited the Pathogen Informatics and Modelling group led by
Dr John Lees through the Theory@EMBL program.
<br><br>

__University of Helsinki, Helsinki, Finland__
: Doctoral researcher, 2018--2022  

: Developed probabilistic models for determining the composition of a
DNA sequencing data sample. Produced efficient open source
implementations of the said models and distributed them. Published
results in scientific journals and presented at international
conferences.
<br><br>

__Helsinki Institute for Information Technology, Helsinki, Finland__
: Research assistant, 2016--2017  

: Designed and implemented models for identifying bacterial strains from DNA sequencing data.

-------

### Other employment {#experience}
&nbsp;
__Heldata oy, Helsinki, Finland__
: Part-time entrepreneur, 2018--2022   

: Planned, organised, and hosted a 250-person data science networking
and recruiting event in 2019 and 2021. Ran the day-to-day affairs of
the business. Worked with customers to find bespoke solutions.
<br><br>

__Stokastiko oy, Helsinki, Finland__
: Independent consultant, 2016--2018  

: Designed and taught a 3-day statistics introductory course for the tax
avoidance unit of the Finnish Tax Administration. Consulted small
businesses and organizations in automating data analysis
pipelines. Founded and operated a one-man company.
<br><br>

-------

### Publications {#publications}
&nbsp;
Peer-reviewed journal articles

: Themisto: a scalable colored _k_-mer index for sensitive pseudoalignment against
hundreds of thousands of bacterial genomes  
JN Alanko, J Vuohtoniemi, __T M&auml;klin__, Simon J Puglisi  
*Bioinformatics* (2023), doi: [10.1093/bioinformatics/btad233](https://doi.org/10.1093/bioinformatics/btad233)

: Strong pathogen competition in neonatal gut colonisation  
__T M&auml;klin__, HA Thorpe, AK P&ouml;ntinen, RA Gladstone, Y Shao, M Pesonen, A McNally, PJ Johnsen, &Oslash; Samuelsen, TD Lawley, A Honkela, and J Corander  
*Nature communications* (2022), doi: [10.1038/s41467-022-35178-5](https://doi.org/10.1038/s41467-022-35178-5)

: Genomic and phenotypic characterization of Clostridium botulinum isolates from an infant botulism case suggests adaptation signatures to the gut  
FP Douillard, Y Derman, C Woudstra, K Selby, __T M&auml;klin__, MB Dorner, H Sax&eacute;n, BG Dorner, H Korkeala, M Lindstr&ouml;m  
*mBio* (2022), doi: [10.1128/mbio.02384-21](https://doi.org/10.1128/mbio.02384-21)

: Bacterial genomic epidemiology with mixed samples  
__T M&auml;klin__, T Kallonen, JN Alanko, &Oslash; Samuelsen, K Hegstad, V M&auml;kinen, J Corander, E Heinz, A Honkela  
*Microbial genomics* (2021), doi: [10.1099/mgen.0.000691](https://doi.org/10.1099/mgen.0.000691.)

: High-resolution sweep metagenomics using ultrafast read mapping and inference  
__T M&auml;klin__, T Kallonen, S David, CJ Boinett, B Pascoe, G M&eacute;ric, DM Aanensen, EJ Feil, S Baker, J Parkhill, SK Sheppard, J Corander, A Honkela  
*Wellcome open research* (2021), doi: [10.12688/wellcomeopenres.15639.2](https://doi.org/10.12688/wellcomeopenres.15639.2)

: Closing _Clostridium botulinum_ Group III genomes using long-read sequencing  
C Woudstra, __T M&auml;klin__, Y Derman, L Bano, H Skarin, C Mazuet, A Honkela, M Lindstr&ouml;m  
*Microbiology resource announcements* (2021), doi: [10.1128/MRA.01364-20](https://doi.org/10.1128/MRA.01364-20)
<br><br>

Book chapters

: Identifying bacterial strains from sequencing data  
__T M&auml;klin__, J Corander, A Honkela  
*Data mining for systems biology* (2018), doi: [10.1007/978-1-4939-8561-6_1](https://doi.org/10.1007/978-1-4939-8561-6_1)

-----

### Grants {#grants}
&nbsp;
: __Theory@EMBL visitor fellowship__  
For visit costs, approx. &pound;5000, 2023

: __DONASCI thesis finalization grant__  
Covered six months salary, 2021

-----

<!-- ### Teaching experience {#teaching} -->

### Service {#service}
Peer review

: 2023 *Genome biology, Nature communications, Microbiome, Scientific reports*  
2022 *BMC microbiology*  
2021 *International conference on machine learning (ICML)*  
2019 *International conference on machine learning (ICML)*
<br><br>


Boards and committees

: Doctoral school in natural sciences, University of Helsinki  
Board member, 2020--2021

: Doctoral programme in Computer Science, University of Helsinki  
Board member, 2020--2021

-----

### Research visits {#researchvisits}
&nbsp;
: __European Bioinformatics Institute, Hinxton, Cambridge, UK__  
Visiting postdoctoral fellow, Jan 2023--July 2023

-----

### External affiliations {#affiliations}
&nbsp;
: __Wellcome Sanger Institute, Hinxton, Cambridge, UK__  
External postdoctoral fellow (contingent worker), Pathogens and microbes programme, 2022-2024

-----

### Research software
&nbsp;
: __mGEMS__  
*Main contributor*  
Taxonomic binner with lineage-level resolution, [PROBIC/mGEMS](https://github.com/PROBIC/mGEMS)


: __mSWEEP__  
*Main contributor*  
Taxonomic profiler with lineage-level resolution, [PROBIC/mSWEEP](https://github.com/PROBIC/mSWEEP)

: __Themisto__  
*With Jarno N Alanko & others*  
Pseudoaligner and _k_-mer index for short-read sequencing data, [algbio/Themisto](https://github.com/algbio/Themisto)

: __ska.rust__  
*With John Lees & others*  
Sequence alignment using split _k_-mers, [bacpop/ska.rust](https://github.com/bacpop/ska.rust)

: __cpprate__  
*Main contributor*  
Variable selection in Bayesian regression models with relative centrality, [tmaklin/cpprate](https://github.com/tmaklin/cpprate)

-----

### Media coverage {#mediacoverage}
Newspapers with national reach

: Sairaalabakteeri ei tartu terveisiin vastasyntyneisiin, kertoo tutkimus  
*Helsingin Sanomat*, Helsinki, Finland, 8 February 2023

-----

### Positions of trust {#positionsoftrust}
&nbsp;
: __University of Helsinki, Helsinki, Finland__  
Shop steward, 2023--2024

-----

### Volunteer experience {#volunteering}
&nbsp;
: __Finnish union of university researchers and teachers__  
Deputy board member, 2023

: __Helsinki university researchers and teachers association__  
Vice chair, 2023 &SmallCircle; Board member, 2022--2023 &SmallCircle; Deputy board member 2020--2021

: __University of Helsinki PhD students association__  
Chair, 2020--2021 &SmallCircle; Secretary 2022

: __Finnish statistical society__  
Secretary, 2017--2022 &SmallCircle; Deputy board member 2022--2023

-----

### Presentations and conference activity {#presentations}
&nbsp;
Session chair

: Probabilistic modelling in microbial ecology workshop  
Kellogg College, University of Oxford, Oxford, UK, March 2023  
<br><br>

Invited talks

: Computational methods in AMR research  
University of Turku, Turku, Finland, October 2023

: Lawley group meeting  
Wellcome Sanger Institute, Hinxton, Cambridge, UK, June 2023

: Theory@EMBL seminar  
European Molecular Biology Laboratory EMBL, Hinxton, Cambridge, UK, June 2023

: Heinz group meeting  
Liverpool school of tropical medicine, Liverpool, UK, February 2023

: Lahti group meeting  
University of Turku, Turku, Finland, January 2023

: AMR-bridge annual meeting  
AMR-bridge consortium, Troms&oslash;, Norway, June 2022

: Thomson group meeting  
Wellcome Sanger Institute, Hinxton, Cambridge, UK, June 2020

: CompLifeSci 4th annual meeting  
&Aring;bo Akademi, Turku, Finland, November 2019
<br><br>

Contributed talks

: Microbial (gen)omics meeting  
Wellcome Sanger Institute, Hinxton, Cambridge, UK, July 2023

: EBI research retreat  
European Bioinformatics Institute, Hinxton, Cambridge, UK, July 2023

: ISMB/ECCB 2019  
Basel, Switzerland, July 2019

: Bioinformatics research and education workshop  
Max Planck Institute for Molecular Genetics, Berlin, Germany, May 2019

: Mathematical and statistical aspects of molecular biology workshop  
European Bioinformatics Institute, Hinxton, Cambridge, UK, April 2019

: Probabilistic modelling in genomics  
Cold Spring Harbor Laboratory, Cold Spring Harbor, NY, US, November 2018

: Regional workshop of European young researchers in Statistics  
Henri Poincar&eacute; Institute, Paris, France, November 2018

: Mathematical and statistical aspects of molecular biology workshop  
University of St Andrews, St Andrews, Scotland, UK, March 2018

-----

### Contact {#footer}
Tommi M&auml;klin -- <script src="/js/contact_me.js"></script><noscript>[Turn on JavaScript to see my email address.]</noscript> -- <script src="/js/call_me.js"></script><noscript>[Turn on JavaScript to see my phone number.]</noscript> -- [<i class="fab fa-linkedin-in fa-1x text-muted"></i>](https://linkedin.com/in/tmaklin)

------
